src/backend/vulkan/src/lib.rs

```diff
@@ -647,6 +649,285 @@ impl hal::Instance<Backend> for Instance {
+
+        match unsafe{display_extension.get_physical_device_display_plane_properties(display.adapter.physical_device.handle)}
+        {
+            Ok(planes_properties)=>{
+                let mut planes = Vec::new();
+                for index in 0..planes_properties.len()
+                {
+                    let compatible_displays = match unsafe{display_extension.get_display_plane_supported_displays(display.adapter.physical_device.handle,index as u32)}
+                    {
+                        Ok(compatible_displays)=>compatible_displays,
+                        Err(error)=>{
+                            match error
+                            {
+                                ash::vk::Result::ERROR_OUT_OF_HOST_MEMORY=>return Err(hal::device::OutOfMemory::Host),
+                                ash::vk::Result::ERROR_OUT_OF_DEVICE_MEMORY=>return Err(hal::device::OutOfMemory::Device),
+                                _=>panic!("Unexpected error returned")
+                            };
+                        }
+                    };
+                    if compatible_displays.contains(&display.handle.0)
+                    {
+                        planes.push(hal::display::Plane{
+                            adapter: &display.adapter,
+                            handle: index as u32,
+                            z_index: planes_properties[index].current_stack_index
+                        });
+                    }
+                }
+                return Ok(planes);
+            }
+            Err(error)=>{
+                match error
+                {
+                    ash::vk::Result::ERROR_OUT_OF_HOST_MEMORY=>return Err(hal::device::OutOfMemory::Host),
+                    ash::vk::Result::ERROR_OUT_OF_DEVICE_MEMORY=>return Err(hal::device::OutOfMemory::Device),
+                    _=>panic!("Unexpected error returned")
+                };
+            }
+        }
+    }
+
+    fn enumerate_builtin_display_modes<'a>(&self,display: &'a hal::display::Display<'a,Backend>,)->Result<Vec<hal::display::DisplayMode<'a,Backend>>,hal::device::OutOfMemory>
+    {
+        let display_extension = ash::extensions::khr::Display::new(&self.entry,&self.raw.inner);
+
+        Ok(match unsafe{display_extension.get_display_mode_properties(display.adapter.physical_device.handle,display.handle.0)}
+        {
+            Ok(display_modes)=>display_modes,
+            Err(error)=>{
+                match error
+                {
+                    ash::vk::Result::ERROR_OUT_OF_HOST_MEMORY=>return Err(hal::device::OutOfMemory::Host),
+                    ash::vk::Result::ERROR_OUT_OF_DEVICE_MEMORY=>return Err(hal::device::OutOfMemory::Device),
+                    _=>panic!("Unexpected error returned")
+                };
+            }
+        }.iter().map(|display_mode_properties|
+        {
+            hal::display::DisplayMode {
+                display: &display,
```

Comment:
The `DIsplay` member is used mainly in the `create_display_plane_surface` to make some tests end eventually throw `DisplayPlaneSurfaceError::UnsupportedParameters`. If you prefer i could remove the tests and replace the `Display` member with `PhysicalDevice`: it is needed for the vulkan functions (like the one in `create_display_plane`) and it is useful to avoid physical device/display.